### PR TITLE
Added simple gui with framerate graph. 

### PR DESCRIPTION
**Pull request by Drenata**
Also let the demo button stay from example code so that people can test GUI features.

The graph freaks out when you turn off vsync. This is because there is a fixed amount of datapoints. The datapoints get updated so quickly that it is hard to read the graph.

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -1,19 +1,24 @@
-
-//#include <glad/glad.h>
 #include "glad.h"
 #include <GLFW/glfw3.h>
 #include <stdlib.h>
 #include <stdio.h>
+#include "imgui.h"
+#include "imgui_impl_glfw_gl3.h"
+
+#define COUNT_OF(x) ((sizeof(x)/sizeof(0[x])) / ((size_t)(!(sizeof(x) % sizeof(0[x])))))
```
Comment (artog) [inline src/main.cpp]: `0[x]` wtf, magic :D

Comment (Drenata) [reply src/main.cpp]: https://stackoverflow.com/questions/4415524/common-array-length-macro-for-c

File: src/main.cpp [hunk 2]
```diff
@@ -28,26 +33,64 @@ int main(void)
         glfwTerminate();
         exit(EXIT_FAILURE);
     }
-    glfwSetKeyCallback(window, key_callback);
+   
     glfwMakeContextCurrent(window);
 	gladLoadGLLoader((GLADloadproc)glfwGetProcAddress);
-    glfwSwapInterval(1);
-    // NOTE: OpenGL error checks have been omitted for brevity
```
Comment (artog) [inline src/main.cpp]: The OpenGL error checks should probably be "un-omitted" and this is a good opportunity for it :)

Comment (Drenata) [reply src/main.cpp]: I'll create a new branch and pull request for it!

**Pull request merged**